link-check/linkCompare.js

```diff
@@ -0,0 +1,151 @@
+import fs from 'fs';
+import { fetch } from '@adobe/fetch';
+import { docx2md } from '@adobe/helix-docx2md';
+
+/**
+ * Compares two links and checks if they have the same host and pathname.
+ *
+ * @param {string} link1 - The first link to compare.
+ * @param {string} link2 - The second link to compare.
+ * @returns {boolean} - Returns true if the links have the same host and pathname, otherwise false.
+ */
+export function compareLink(link1, link2) {
+  const url1 = new URL(link1.trim(), 'https://business.adobe.com/');
+  const url2 = new URL(link2.trim(), 'https://business.adobe.com/');
+
+  return (url1.host === url2.host) && (url1.pathname === url2.pathname);
+}
+
+/**
+ * Retrieves the content from a source.
+ *
+ * @param {string} source - The source URL or file path.
+ * @param {Function} fetchFn - The function used to fetch the content from the source.
+ * @returns {Promise<string>} - A promise that resolves to the content of the source.
+ */
+async function getContent(source, fetchFn) {
```

Comment:
Dependency injection, makes testing easier since you cant mock a module, especially in Mocha.

Reply:
Could you use the global `fetch` we are importing from @adobe/fetch and overwrite the global?
```
  beforeEach(() => {
    window.fetch = stub().returns(htmlResponse());
  });

  afterEach(() => {
    window.fetch = ogFetch;
  });
```

I know since this is pure node, we don't have the window object, but I do agree that adding it as a param for testing purposes is probably not the best practice.

Reply:
Sorry, I would argue that globals are not a good practice and dependency injection is pretty established. Normally it is part of a class constructor or a config object param.
This is node and we can not overwrite an import, like with import maps, as we don't have a browser.

Reply:
Dependency injection can have benefits outside of testing as you can adjust fetch outside of the method.
For example you could add an abort or timeout or set cache setting.
```js
const signal = timeoutSignal(1000);
const fetchFn =  (url) => fetch(url, { signal, cache: 'no-store' });

getContent(source, fetchFn);
```
Or use a different fetch with a custom user agent.
```js
import { context } from '@adobe/fetch';
const { fetch } = context({ userAgent: 'custom-fetch' });

getContent(source, fetch);
```

Reply:
I get your point and I do think dependency injection is a best practice, but since you often pass the default global fetch in the function params, it seems as though the param is there for testing. It makes sense that we can't override the import, but maybe we can uses [sinon](https://sinonjs.org/how-to/stub-dependency/) or [chai-fetch](https://www.chaijs.com/plugins/chai-fetch/)? 

Using these stubs would follow the best practices set [out in this discussion](https://github.com/orgs/adobecom/discussions/814#discussioncomment-6060759), though Okan's examples are using the browser.

If you think either sinon or chai-fetch to stub would be good to do eventually, but maybe too much rework for this ticket, we can create an issue against the repo. This ticket already has a lot of good work and good changes we want to get merged in.

Reply:
Discussed in meeting.
We don't normally add parameters just for testing but there is no other way to pass / use the stub without some intense hacks. Also adding fetch as a parameter gives benefits like being able to set a timeout from outside of the method so there is some utility beyond the test. The discussion best practices don't work in node, apples & oranges, and this is not the same fetch function as window.fetch.